### PR TITLE
POST and start top up flow in rapidpro

### PR DESCRIPTION
**Pull request by DevChima**

### CODE REVIEW

File: ada/tasks.py [hunk 1]
```diff
@@ -22,3 +26,21 @@ def submit_whatsappid_to_rapidpro(whatsappid):
             flow=settings.ADA_PROTOTYPE_SURVEY_FLOW_ID,
             urns=[f"whatsapp:{whatsappid.lstrip('+')}"],
         )
+
+
+def submit_whatsappid_to_rapidpro_topup(whatsappid):
```
Comment (erikh360) [inline ada/tasks.py]: Missing the `@app.task(` decorator for these two tasks

File: template/topup_request.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <h1>Your topup request has been received.</h1>
```
Comment (erikh360) [inline template/topup_request.html]: How is this view called? not saying its wrong, just not sure why this template is needed.

File: ada/tasks.py (hunk 1 above)
Comment (erikh360) [reply ada/tasks.py]: I'm thinking the two `submit_whatsappid_to_rapidpro*` tasks could be better named, bit confusing.

`start_prototype_survey_flow` and `start_topup_flow` is a bit more descriptive of what they do.

**Review by erikh360 — COMMENTED** [review]
looks good but few comments and Q's

File: template/topup_request.html (hunk 2 above)
Comment (DevChima) [reply template/topup_request.html]: So the user clicks on a Whatsapp link first which then shows this template letting the contact know that their top-up request has been received and then we fire a POST to the endpoint to start the flow. So this view is just for user experience.

File: ada/tasks.py (hunk 1 above)
Comment (DevChima) [reply ada/tasks.py]: I agree.

File: template/topup_request.html (hunk 2 above)
Comment (DevChima) [reply template/topup_request.html]: https://drive.google.com/file/d/1hxLEkEwxdPtDr3pWX-DK9_YonrB3w0PJ/view?usp=sharing

File: ada/tasks.py [hunk 3]
```diff
@@ -15,10 +19,44 @@
...
+    if rapidpro and settings.ADA_TOPUP_FLOW_ID:
+        return rapidpro.create_flow_start(
+            extra={},
+            flow=settings.ADA_TOPUP_FLOW_ID,
+            urns=[f"whatsapp:{whatsappid.lstrip('+')}"],
+        )
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded, TembaHttpError),
```
Comment (erikh360) [inline ada/tasks.py]: TembaHttpError is a rapidpro specific error, not possible here

Comment (DevChima) [reply ada/tasks.py]: Removed

**Review by erikh360 — APPROVED** [review]

**Pull request merged**